openapi-cli/src/main/java/io/ballerina/openapi/generators/client/FunctionSignatureGenerator.java

```diff
@@ -286,22 +289,29 @@ public Node getQueryParameters(Parameter parameter, NodeList<AnnotationNode> par
             paramType = getValidName(extractReferenceType(parameterSchema.get$ref()), true);
         } else {
             paramType = convertOpenAPITypeToBallerina(parameterSchema.getType().trim());
-            if (parameterSchema.getType().equals("number")) {
+            if (parameterSchema.getType().equals(NUMBER)) {
                 if (parameterSchema.getFormat() != null) {
                     paramType = convertOpenAPITypeToBallerina(parameterSchema.getFormat().trim());
                 }
-            }
-            if (parameterSchema instanceof ArraySchema) {
+            } else if (parameterSchema instanceof ArraySchema) {
                 ArraySchema arraySchema = (ArraySchema) parameterSchema;
                 if (arraySchema.getItems().getType() != null) {
                     String itemType = arraySchema.getItems().getType();
-                    if (itemType.equals("string") || itemType.equals("integer") || itemType.equals("boolean")
-                            || itemType.equals("number")) {
+                    if (itemType.equals(STRING) || itemType.equals(INTEGER) || itemType.equals(BOOLEAN)) {
                         paramType = convertOpenAPITypeToBallerina(itemType) + SQUARE_BRACKETS;
+                    } else if (itemType.equals(NUMBER)) {
+                        paramType = convertOpenAPITypeToBallerina
+                                (arraySchema.getItems().getFormat().trim()) + SQUARE_BRACKETS;
+                    } else {
+                        throw new BallerinaOpenApiException("Unsupported parameter type is found in the parameter : " +
+                                parameter.getName());
                     }
                 } else if (arraySchema.getItems().get$ref() != null) {
                     paramType = getValidName(extractReferenceType(
                             arraySchema.getItems().get$ref().trim()), true) + SQUARE_BRACKETS;
+                } else {
+                    throw new BallerinaOpenApiException("Please define the type of the parameter : " +
```

Comment:
Shall we change this with `Please define the array item type of the parameter : ` since here we are considering array types?

Reply:
Updated